cpp/oneapi/dal/data/detail/access_wrapper_host.hpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/data/detail/access_iface_wrapper.hpp"
+#include "oneapi/dal/data/detail/access_iface_type_traits.hpp"
+
+#include <stdexcept> // TODO: change by oneDAL exceptions
+
+namespace oneapi::dal::detail {
+
+template <typename T>
+class access_wrapper_impl_host {
+public:
+    using policy_t = host_seq_policy;
+    using alloc_kind_t = host_only_alloc;
+
+public:
+    access_wrapper_impl_host(T& obj)
+        : obj_(obj) {}
+
+    template <typename Block>
+    void pull_rows(const policy_t&,
+                   Block& block, const row_block& index,
+                   const alloc_kind_t&) const {
+        if constexpr (has_pull_rows_host<T, typename Block::data_t>::value) {
+            obj_.pull_rows(block, index.rows);
+        } else {
+            throw std::runtime_error("pulling rows is not supported");
+        }
+    }
+
+    template <typename Block>
+    void pull_column(const policy_t&,
+                     Block& block, const column_values_block& index,
+                     const alloc_kind_t&) const {
+        if constexpr (has_pull_column_host<T, typename Block::data_t>::value) {
+            obj_.pull_column(block, index.column_index, index.rows);
+        } else {
+            throw std::runtime_error("pulling column is not supported");
```

Comment:
Will this type of exception be changed after the request https://github.com/oneapi-src/oneDAL/pull/676?

Reply:
Yes, I especially marked TODO at the head of the file: https://github.com/oneapi-src/oneDAL/blob/b5810686c9a87fd4b33e9b332aece3e6f96fcc87/cpp/oneapi/dal/data/detail/access_wrapper_host.hpp#L22